app/code/core/Mage/Customer/controllers/AccountController.php

```diff
@@ -895,7 +895,7 @@ public function resetPasswordPostAction()
      */
     protected function getCustomerId()
     {
-        $customerId = $this->getRequest()->getQuery("id");
+        $customerId = $this->getRequest()->getQuery('id', false);
```

Comment:
```suggestion
        $customerId = $this->getRequest()->getQuery('id', '');
```

Would it be better to use empty string instead of false?

With strict types passing bool to strlen results in a fatal error.

Reply:
Lines 904 - 906:
```php
            $customerId = count($customerCollection) === 1
                ? $customerCollection->getFirstItem()->getId()
                : false;
```
So, the default `false` is the expected behavior. If we modify this to empty string, is there a BC concern?

Reply:
Before you change default was `null`, now it 'string' ...

```
    /**
     * @return string|null
     * @throws Mage_Core_Exception
     */
    protected function getCustomerId()
    {
        $customerId = $this->getRequest()->getQuery('id');
        if (!is_null($customerId) && strlen($customerId) > 12) {
            $customerCollection = Mage::getModel('customer/customer')
                ->getCollection()
                ->addAttributeToSelect(['rp_customer_id'])
                ->addFieldToFilter('rp_customer_id', $customerId);
            $customerId = count($customerCollection) === 1
                ? $customerCollection->getFirstItem()->getId()
                : null;
        }

        return $customerId;
    }
```

Reply:
Now you removed `false` as a possible return value.

Reply:
I wanted to avoid antother return type ...  better use false as default and replace `is_null` check with `is_string`?